Add speedscale integration (#1238)
DataDog/integrations-extras: speedscale/README.md

# Agent Check: Speedscale

## Overview

> This integration publishes the results from [Speedscale][1] into Datadog.

Reviewer: Would you be able to add a bit more detail here about the kind of data that Speedscale collects, and what types of data it's sending into Datadog?

Author: Thank you, that section was definitely too vague, I added more details.

## Setup

### Configuration

In order to utilize the integration you need to capture the following from Datadog:

* [API Key][2] - An API key is required by the Datadog Agent to submit metrics and events to Datadog

A best practice is to save this into environment variables like so:

```

> export DDOG_API_KEY=0

Reviewer: Once the user has saved their API key, where would they enter it to connect to Speedscale? It would be helpful to have some step by step instructions here.

Author: Thank you, I reworked this section to keep all the configuration in one place. Right now this is done through continuous integration and that is specifically called out in my latest update.

> # Agent Check: Speedscale

Reviewer: I'd recommend removing this line since the integration is not using the Agent

Author: updated

## Overview

> This integration publishes traffic replay results from [Speedscale][1] into Datadog. This lets you combine your observability data from Datadog with the results of a particular replay to investigate the root cause of poor performance. Find and troubleshoot potential performance issues before they show up in production with a combination of Speedscale and Datadog.

Reviewer: ```suggestion
This integration publishes traffic replay results from [Speedscale][1] into Datadog. This lets you combine your observability data from Datadog with the results of a particular Speedscale replay to investigate the root cause of poor performance. Find and troubleshoot potential performance issues before they show up in production with the Speedscale Datadog integration.
```

## Setup

### Configuration

> 1. To use this integration you need a Datadog [API Key][2] so that the Datadog Agent can submit metrics and events to Datadog.

Reviewer: ```suggestion
1. To use this integration you need a Datadog [API Key][2] so that metrics and events can be submitted into Datadog.
```
Correct me if I'm wrong, but I thought that this integration did not use the Datadog agent?

>    A best practice is to save this into an environment variable. Most likely you will store this environment variable in your continuous integration system, but when doing a one-off test you can access in your terminal like so:

Reviewer: ```suggestion
A best practice is to save this value into an environment variable. Most likely you will store this environment variable in your continuous integration system, but when doing a one-off test you can access it in your terminal like so:
```

> ```

Reviewer: ```suggestion
    ```
```

export DDOG_API_KEY=0
```

> 2. Gather the report ID of a specific report that you want to upload into Datadog. When working with continuous integration, get the report ID associated with your commit hash. Store this report ID in an environment variable:

Reviewer: ```suggestion
2. Gather the report ID of a specific report that you'd like to upload to Datadog. When working with continuous integration, get the report ID associated with your commit hash. Store this report ID in an environment variable:
```

# Agent Check: Speedscale

## Overview

> This integration publishes traffic replay results from [Speedscale][1] into Datadog. This lets you combine your observability data from Datadog with the results of a particular replay to drill-down into the root cause of poor performance. Find and troubleshoot potential performance issues before they show up in production with a combination of Speedscale and Datadog.

Reviewer 2: ```suggestion
This integration publishes traffic replay results from [Speedscale][1] into Datadog. This lets you combine your observability data from Datadog with the results of a particular replay to investigate the root cause of poor performance. Find and troubleshoot potential performance issues before they show up in production with a combination of Speedscale and Datadog.
```

## Setup

### Configuration

> In order to utilize the integration you need to capture the following from Datadog:
> 
> * [API Key][2] - An API key is required by the Datadog Agent to submit metrics and events to Datadog
> 
> A best practice is to save this into an environment variable. Most likely you will store this environment variable in your continuous integration system, but when doing a one-off test you can access in your terminal like so:

Reviewer 2: ```suggestion
1. To use this integration you need a Datadog [API Key][2] so that the Datadog Agent can submit metrics and events to Datadog.

   A best practice is to save this into an environment variable. Most likely you will store this environment variable in your continuous integration system, but when doing a one-off test you can access in your terminal like so:
```

```
export DDOG_API_KEY=0

> ```

Reviewer 2: indent these lines by 3 spaces to nest the codeblock inside the list item.

> Gather the report id of a specific report that you want to upload into Datadog. When working with continuous integration you should be able to get the report id associated with your commit hash. Store this report id in an environment variable:

Reviewer 2: ```suggestion
2. Gather the report ID of a specific report that you want to upload into Datadog. When working with continuous integration, get the report ID associated with your commit hash. Store this report ID in an environment variable:
```

```
export SPD_REPORT_ID=0

> ```

Reviewer 2: indent 3 spaces

> Now with the specific report id and the Datadog API key, you run the `speedctl` CLI to export that traffic replay report as a Datadog event.

Reviewer 2: ```suggestion
3. With the specific report ID and the Datadog API key, run the `speedctl` command to export that traffic replay report as a Datadog event.
```

```
speedctl export datadog report ${SPD_REPORT_ID} --apiKey ${DDOG_API_KEY}
✔ {"status":"ok",...}

> ```

Reviewer 2: indent three spaces. Also, the check mark is causing an error in the file because it is non-ASCII. Is that line part of the command? or what the command returns (if this, then just delete the line).

### Validation

> After you have exported the report, you should see it in the Datadog [Event Stream][2].

Reviewer 2: ```suggestion
View the Datadog [Event Stream][2] to see your exported report.
```

## Data Collected

### Metrics

Speedscale does not include any metrics.

### Service Checks

Speedscale does not include any service checks.

### Events

> The Speedscale integration sends events to your [Datadog Event Stream][3] when a traffic replay is complete to understand the impact this has on your metrics.

Reviewer 2: ```suggestion
The Speedscale integration sends events to your [Datadog Event Stream][3] when a traffic replay is complete to help you understand the impact this has on your metrics.
```
Not sure I understand what "the impact this has on your metrics" means. What's the "this"? Maybe rephrase to be specific?
